test/parallel/test-fs-whatwg-url.js

```diff
@@ -72,3 +72,149 @@ if (common.isWindows) {
     message: `File URL host must be "localhost" or empty on ${os.platform()}`
   }));
 }
+
+// path ending with forward slash are not permitted
+const urlWithSlash = common.isWindows ?
```

Comment:
Can you add some tests for the string path and the buffer paths? It might worth putting these tests into a new file and reuse the test cases. Something like

```js
function runTests(withSlash, withoutSlash) {
  ...
}

runTests(urlWithSlash, urlWithoutSlash);
runTests(stringWithSlash, stringWithoutSlash);
runTests(bufferWithSlash, bufferWithoutSlash);
```